witness/src/Dscp/Witness/Logic/Getters.hs

```diff
@@ -122,16 +122,16 @@ resolveNext = SD.queryOne . NextBlockOf . headerHash >=> pure . map unNextBlock
 -- TODO [DSCP-367] Unite these two below
 -- | Safely get an account.
 getAccountMaybe
-    :: WitnessWorkMode ctx m
+    :: (WitnessWorkMode ctx m, WithinReadSDLock)
     => Address -> SdReadM 'ChainOnly m (Maybe Account)
 getAccountMaybe =
-    lift . runStateSdMLocked . SD.queryOne . AccountId
+    lift . runStateSdM . SD.queryOne . AccountId
 
 -- | Safely get an account taking mempool into consideration.
 getMempoolAccountMaybe
-    :: WitnessWorkMode ctx m
+    :: (WitnessWorkMode ctx m, WithinReadSDLock)
     => Address -> SdReadM 'ChainAndMempool m (Maybe Account)
-getMempoolAccountMaybe addr = readingSDLock $ liftSdM $ SD.queryOne (AccountId addr)
+getMempoolAccountMaybe addr = liftSdM $ SD.queryOne (AccountId addr)
```

Comment:
Do I understand correctly that every time those operations were called in existing code, the lock has been already taken, and therefore taking it again were obsolete?

Reply:
Not that far ago every small operation took lock itself locally, including this one.
When I modified Explorer API recently, I mostly replaced local locking with endpoint-level locking, but yes, I missed this function, it took the reading lock again for nothing until this PR.